### ISSUE TITLE
Deploying hyperledger fabric 2.2

### ISSUE THREAD

**Issue by nkaramolegos** [litong01/fabric-operator#1]
Can I use this project to deploy and maintain HLF 2.2? In which file can I set the HLF network architecture?

**Comment by litong01** [comment]
@nkaramolegos this uses non-system channel feature at present. You can certainly try it.

**Comment by nkaramolegos** [comment]
I am not sure that I understand your answer.

**Comment by nkaramolegos** [comment]
Can you give some more details on how can I test this operator? I am going to build a single host cluster using kind or kubeadm. Do you have any proposal on that? Which is better?

**Comment by litong01** [comment]
You probably can see a lot of sample deployment yaml files from config/samples directory. That directory contains yaml files allow you to setup ca, peer,orderer nodes. Also there is this Agent operator which you can use to perform fabric network operations such as create channel, join org, join channel, signoff, commit etc. As you go about this, if you can also write up some docs and contribute, that will be great.

Also there is this molecule directory which contains a lot of scenarios to setup multiple orgs and perform a lot of operations. Take a look at this if you are interested

To run these scenarios, you can setup your env to have ansible, molecule and few python packages, then you will do the following:

```
1. mol converge -s kind
2. mol converge -s systemchannel

```

The first step is to setup kind k8s cluster, the 2nd is to deploy a two org network and create channel, join org, join peer, etc.

**Comment by litong01** [comment]
Yes. You absolutely can do that and the molecule scenarios provide some great examples.

**Comment by nkaramolegos** [comment]
I see. Thank you for your advice. I am a bit confused, why do I need the operator? what is the difference from deploying just the network using k8s (just deployment of the fabric images in a cluster)? I am new to k8s and I am trying to understand the usage (thus the advantages) of an operator and particularity this operator.

**Comment by litong01** [comment]
Probably read a bit about what k8s operator is and why and how it is developed and used. You may google search for k8s operator.

**Comment by nkaramolegos** [comment]
Can you please me give more direction on how should I install and configure ansible, openshift and molecule on ubuntu in order to start? I have problems understanding how exactly should I configure these tools
